src/oas2/service.ts

```diff
@@ -58,13 +58,28 @@ export const transformOas2Service: Oas2HttpServiceTransformer = ({ document }) =
       if (!sec) return null;
 
       return keys(sec).map(key => {
-        return securitySchemes.find(securityScheme => {
-          return securityScheme.key === key;
-        });
+        const ss = securitySchemes.find(securityScheme => securityScheme.key === key);
+        if (ss && ss.type === 'oauth2') {
+          return {
+            ...ss,
+            flows: mapValues(ss.flows, (flow: IOauth2Flow) => ({
+              ...flow,
+              scopes: pickBy(flow.scopes, (_val: string, scopeKey: string) => {
+                const secKey = sec[key];
+                if (secKey) return secKey.includes(scopeKey);
+                return false;
+              }),
+            })),
+          };
+        }
+
+        return ss;
```

Comment:
Is this the same block of code that's in the oas3 service? Could we break it out to a common function to use in both spots?